src/payment/strategies/paypal-commerce/paypal-commerce-payment-processor.ts

```diff
@@ -0,0 +1,67 @@
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+
+import { OverlayPaypal } from './index';
+
+const widthWindow = 450;
+const heightWindow = 600;
+
+export default class PaypalCommercePaymentProcessor {
+    private _window = window;
+    private _popup?: Window | null;
+
+    constructor(
+        private _overlay: OverlayPaypal
+    ) {}
+
+    paymentPayPal(approveUrl: string): Promise<any> {
+        return new Promise((resolve, reject) => {
+            const closeWindow = (isResolve: boolean) => {
+                this._overlay.remove();
+
+                if (this._popup) {
+                    this._popup.close();
+                }
+
+                isResolve
+                    ? resolve(true)
+                    : reject(new MissingDataError(MissingDataErrorType.MissingPayment));
+            };
+
+            const handlerMessage = (event: MessageEvent) => {
+                if (event.origin.includes('sandbox.paypal.com')) {
+                    const data = JSON.parse(event.data);
+
+                    if (data.operation === 'return_to_merchant' && data.updateParent) {
+                        this._window.removeEventListener('message', handlerMessage);
+                        closeWindow(true);
+                    }
+                }
+            };
+
+            this._window.addEventListener('message', handlerMessage);
+
+            const params = `
```

Comment:
🍺 I think you could move this variable outside the class like `widthWindow` variable.